server/src/main/java/io/unitycatalog/server/security/SecurityContext.java

```diff
@@ -0,0 +1,126 @@
+package io.unitycatalog.server.security;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.unitycatalog.server.UnityCatalogServer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import lombok.Getter;
+import lombok.SneakyThrows;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SecurityContext {
+
+  public interface Issuers {
+    String INTERNAL = "internal";
+  }
+
+  private static final ObjectMapper mapper = new ObjectMapper();
+  private static final Logger LOGGER = LoggerFactory.getLogger(UnityCatalogServer.class);
+
+  private final Path certsFile;
+  private final Path serviceTokenFile;
+
+  @Getter private final RSAPublicKey rsaPublicKey;
+  @Getter private final RSAPrivateKey rsaPrivateKey;
+  @Getter private final Algorithm algorithm;
+  @Getter private final String serviceToken;
+  @Getter private final String keyId;
+  @Getter private final String serviceName;
+  @Getter private final String localIssuer;
+
+  @SneakyThrows
+  public SecurityContext(
+      Path configurationFolder,
+      SecurityConfiguration securityConfiguration,
+      String serviceName,
+      String localIssuer) {
+    this.serviceName = serviceName;
+    this.localIssuer = localIssuer;
+
+    certsFile = configurationFolder.resolve("certs.json");
+    serviceTokenFile = configurationFolder.resolve("token.txt");
+
+    rsaPublicKey = securityConfiguration.rsaPublicKey();
+    rsaPrivateKey = securityConfiguration.rsaPrivateKey();
+    algorithm = securityConfiguration.algorithmRSA();
+    keyId = securityConfiguration.getKeyId();
+
+    // Do this every time because the SecurityConfiguration parameters
+    // might have changed since last started.
+    serviceToken = createServiceToken();
```

Comment:
to clarify: does this mean every time the service restarts the tokens that were previously issued would be invalid?

Reply:
The previous token(s) only become invalid when the signing keys are rotated, but yes this does create a new token on startup.